### ISSUE TITLE
RichTextBox could not be used in VS designer with AOT enabled

### ISSUE THREAD

**Issue by ismlsmile** [dotnet/winforms#8730]
### Environment

17.5.0

### .NET version

7.0.102

### Issue description

when drag a RichTextBox to a Form, VS Designer will show this error:
Built-in COM has been disabled via a feature switch. See https://aka.ms/dotnet-illink/com for more

![image](https://user-images.githubusercontent.com/1876179/222307893-d9791b14-d135-43a1-94d7-2161fe76c1ed.png)

demo project:
[TestWinFormsApp.zip](https://github.com/dotnet/winforms/files/10866768/TestWinFormsApp.zip)

### Steps to reproduce

1. edit csproj file , add:
	<PublishAot>true</PublishAot>
	<_SuppressWinFormsTrimError>true</_SuppressWinFormsTrimError>

2. drag a RichTextBox to a Form

**Comment by RussKie** [comment]
.NET Desktop - this includes Windows Forms and WPF - does not work with COM disabled and do not support trimming or NativeAOT.

**Comment by merriemcgaw** [comment]
Correct, we simply cannot work properly with COM disabled and are not set up to enable trimming and NativeAOT.
